crates/dataflow-jit/src/ir/types.rs

```diff
@@ -0,0 +1,393 @@
+use crate::{
+    codegen::NativeType,
+    ir::{function::InputFlags, LayoutId, RowLayoutCache},
+};
+use bitvec::vec::BitVec;
+use std::fmt::{self, Debug, Display, Write};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ColumnType {
+    Bool,
+    U16,
+    I16,
+    U32,
+    I32,
+    U64,
+    I64,
+    F32,
+    F64,
+    /// Represents the days since Jan 1 1970 as an `i32`
+    Date,
```

Comment:
It seems odd to have `Date` and `Timestamp` as primitives in a low-level IR. Is there a reason they cannot be represented with integers?

Reply:
They totally could be, this just somewhat simplifies handling since it's well-known to the compiler